project-base/storefront/components/Pages/Order/TransportAndPayment/TransportAndPaymentContent.tsx

```diff
@@ -120,58 +95,122 @@ export const TransportAndPaymentContent: FC<TransportAndPaymentContentProps> = (
-                <ErrorPopup
-                    onCloseCallback={() => setErrorPopupVisibility(false)}
-                    fields={transportAndPaymentValidationMessages}
-                    gtmMessageOrigin={GtmMessageOriginType.transport_and_payment_page}
-                />
-            )}
+            <OrderLayout
+                activeStep={2}
+                isTransportOrPaymentLoading={Boolean(isTransportSelectionLoading) || isPaymentSelectionLoading}
+                isLoading={isTransportAndPaymentSkeletonVisible}
+            >
+                {isTransportAndPaymentSkeletonVisible ? (
+                    <Skeleton className="h-64 w-full" />
+                ) : (
+                    <>
+                        {!!transportsData?.transports.length && (
+                            <TransportAndPaymentSelect
+                                transports={transportsData.transports}
+                                lastOrderTransportUuid={lastOrderData?.lastOrder?.transport.uuid ?? null}
+                                lastOrderPaymentUuid={lastOrderData?.lastOrder?.payment.uuid ?? null}
+                                lastOrderPickupPlace={lastOrderPickupPlace}
+                                changeTransportInCart={changeTransportInCart}
+                                changePaymentInCart={changePaymentInCart}
+                                isTransportSelectionLoading={isTransportSelectionLoading}
+                            />
+                        )}
+                        <OrderAction
+                            buttonBack={t('Back')}
+                            buttonNext={t('Contact information')}
+                            hasDisabledLook={
+                                hasValidationErrors(validationMessages) ||
+                                isTransportSelectionLoading ||
+                                isPaymentSelectionLoading
+                            }
+                            isLoading={
+                                (isTransportSelectionLoading || isPaymentSelectionLoading) &&
+                                transport !== null &&
+                                payment !== null
+                            }
+                            withGapTop
+                            withGapBottom
+                            buttonBackLink={cartUrl}
+                            nextStepClickHandler={onSelectTransportAndPaymentHandler}
+                        />
+                        {isErrorPopupVisible && (
+                            <ErrorPopup
+                                onCloseCallback={() => setErrorPopupVisibility(false)}
+                                fields={validationMessages}
+                                gtmMessageOrigin={GtmMessageOriginType.transport_and_payment_page}
+                            />
+                        )}
+                    </>
+                )}
+            </OrderLayout>
         </>
     );
 };
+
+const getLastOrderPickupPlace = (
```

Comment:
Can we move these functions into `helpers.tsx`?

Reply:
We can, moved.